internal/security/secretstore/init.go

```diff
@@ -340,35 +340,40 @@ func (b *Bootstrap) BootstrapHandler(ctx context.Context, _ *sync.WaitGroup, _ s
 		os.Exit(1)
 	}
 
-	cert := NewCerts(req, configuration.SecretService.CertPath, rootToken, configuration.SecretService.GetSecretSvcBaseURL(), lc)
-	existing, err := cert.AlreadyinStore()
-	if err != nil {
-		lc.Error(err.Error())
-		os.Exit(1)
-	}
+	// Uploading Kong's TLS certificate is optional if cert's path is not present
+	if len(configuration.SecretService.CertPath) > 0 {
+		cert := NewCerts(req, configuration.SecretService.CertPath, rootToken, configuration.SecretService.GetSecretSvcBaseURL(), lc)
+		existing, err := cert.AlreadyinStore()
+		if err != nil {
+			lc.Error(err.Error())
+			os.Exit(1)
+		}
 
-	if existing == true {
-		lc.Info("proxy certificate pair are in the secret store already, skip uploading")
-		return false
-	}
+		if existing {
+			lc.Info("proxy certificate pair are in the secret store already, skip uploading")
+			return false
+		}
 
-	lc.Info("proxy certificate pair are not in the secret store yet, uploading them")
-	cp, err := cert.ReadFrom(configuration.SecretService.CertFilePath, configuration.SecretService.KeyFilePath)
-	if err != nil {
-		lc.Error("failed to get certificate pair from volume")
-		os.Exit(1)
-	}
+		lc.Info("proxy certificate pair are not in the secret store yet, uploading them")
+		cp, err := cert.ReadFrom(configuration.SecretService.CertFilePath, configuration.SecretService.KeyFilePath)
+		if err != nil {
+			lc.Error("failed to get certificate pair from volume")
+			os.Exit(1)
+		}
 
-	lc.Info("proxy certificate pair are loaded from volume successfully, will upload to secret store")
+		lc.Info("proxy certificate pair are loaded from volume successfully, will upload to secret store")
 
-	err = cert.UploadToStore(cp)
-	if err != nil {
-		lc.Error("failed to upload the proxy cert pair into the secret store")
-		lc.Error(err.Error())
-		os.Exit(1)
-	}
+		err = cert.UploadToStore(cp)
+		if err != nil {
+			lc.Error("failed to upload the proxy cert pair into the secret store")
+			lc.Error(err.Error())
+			os.Exit(1)
+		}
 
-	lc.Info("proxy certificate pair are uploaded to secret store successfully, Vault init done successfully")
+		lc.Info("proxy certificate pair are uploaded to secret store successfully, Vault init done successfully")
+	} else {
+		lc.Info("No proxy certificate pair to upload, Vault init done successfully")
+	}
```

Comment:
Better check to see if this will generate a merge conflict with @beaufrusetta's change.  Probably better to get @beaufrusetta change in first, and see if this section still need to be updated.  Can follow up later with a PR to eliminate the cert uploading to Vault altogether.

Reply:
This is definitely going to conflict, especially since I have a pending update to add an else statement with a log entry...and at the end of that block I have a closing log entry that states vault init complete.

Reply:
@beaufrusetta , we'll land your PR first and then @jim-wang-intel can rebase and resolve conflicts in his PR.

Reply:
rebase and resolved the conflicts, thanks guys.